worker-document/src/main/java/com/hpe/caf/worker/document/tasks/DocumentTask.java

```diff
@@ -126,13 +126,27 @@ protected WorkerResponse createWorkerResponseImpl()
         // If the response message includes any scripts then it is in the v2 message format
         final int resultMessageVersion = (documentWorkerResult.scripts == null) ? 1 : 2;
 
+        //get the setting enable exception on failure
+        final Boolean enableExceptionOnFailure = application.getConfiguration().getEnableExceptionOnFailure();
+        if(enableExceptionOnFailure==true) {
+            if(ChangeLogFunctions.hasFailures(changes)){
```

Comment:
From a runtime perspective `ChangeLogFunctions.hasFailures(changes)` has already been called.  We need to reorganize so that it is not called twice.